src/main/java/spotify/playback/data/lyrics/GeniusLyricsScraper.java

```diff
@@ -47,37 +58,96 @@ public String getSongLyrics(String artistName, String songName) {
+
+    String searchUrl = "https://genius.com/api/search?q=" + processedArtistName + " " + processedSongName;
+    System.out.println(searchUrl);
+
     Connection.Response response = Jsoup.connect(searchUrl)
-      .userAgent(USER_AGENT)
-      .ignoreContentType(true)
-      .execute();
+            .userAgent(USER_AGENT)
+            .ignoreContentType(true)
+            .execute();
     String json = response.body();
 
-    String matchingPath = findMatchingPath(json, artistName, songName);
+    String matchingPath = findMatchingPath(json, processedArtistName, processedSongName);
+    System.out.println(matchingPath);
+
     if (matchingPath != null) {
       return "https://genius.com" + matchingPath;
     }
     return null;
   }
 
+  private String preprocessString(String input) {
+    // Remove all occurrences of problematic symbols
+    input = input.replace("#", "");
+    input = input.replace("@", "");
+    input = input.replace("_", "");
+
+    // Check if the string contains only non-language characters outside the brackets
+    String outsideBrackets = input.replaceAll("\\(.*?\\)", "").trim();
+    if (outsideBrackets.isEmpty() || outsideBrackets.matches("[^\\p{L}\\p{N}]+")) {
+      // If so, keep the content inside the brackets
+      Pattern pattern = Pattern.compile("\\((.*?)\\)");
+      Matcher matcher = pattern.matcher(input);
+      if (matcher.find()) {
+        return matcher.group(1).trim();
+      }
+    }
+
+    // Remove content inside brackets
+    input = input.replaceAll("\\(.*?\\)", "").trim();
+
+    // Remove any trailing text after an unpaired opening bracket
+    if (input.contains("(") && !input.contains(")")) {
+      input = input.substring(0, input.indexOf('(')).trim();
+    }
+
+    return input;
+  }
+
   private String findMatchingPath(String json, String artistName, String songName) {
     JsonObject jsonObject = gson.fromJson(json, JsonObject.class);
     JsonArray hits = jsonObject.getAsJsonObject("response").getAsJsonArray("hits");
 
+    // Normalize the search strings
+    String normalizedArtistName = normalizeString(artistName);
+    String normalizedSongName = normalizeString(songName);
+    String[] artistNameParts = normalizedArtistName.split("\\s+");
+
     for (JsonElement hit : hits) {
       JsonObject result = hit.getAsJsonObject().getAsJsonObject("result");
       String artistNames = result.get("artist_names").getAsString();
       String title = result.get("title").getAsString();
 
-      if (StringUtils.startsWithIgnoreCase(artistNames, artistName) && StringUtils.startsWithIgnoreCase(title, songName)) {
+      // Normalize the JSON strings
+      String normalizedArtistNames = normalizeString(artistNames);
+      String normalizedTitle = normalizeString(title);
+
+      // Check for partial matches in artist names
+      boolean artistMatch = Arrays.stream(artistNameParts)
+              .allMatch(part -> normalizedArtistNames.contains(part));
```

Comment:
This is just syntactic sugar, but please write it like this:

        .allMatch(normalizedArtistNames::contains);